commit 8ef1f1a717cc
Author: Andrew McCreight <continuation@gmail.com>
Date:   Thu Jan 4 14:59:14 2018 -0800

    Bug 1426801 - Clear gFirstIdleTask if we fail to dispatch the runnable. r=mrbkap, a=gchang
    
    MozReview-Commit-ID: 23f9VDTyyqg
    
    --HG--
    extra : source : 1eb22728208e6fc9c785e5cf11bb68929d0a555d
---
 dom/ipc/ContentChild.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git dom/ipc/ContentChild.cpp dom/ipc/ContentChild.cpp
index cc7769daa805..84881a36f04d 100644
--- dom/ipc/ContentChild.cpp
+++ dom/ipc/ContentChild.cpp
@@ -1875,7 +1875,10 @@ ContentChild::RecvPBrowserConstructor(PBrowserChild* aActor,
     MOZ_ASSERT(!gFirstIdleTask);
     RefPtr<CancelableRunnable> firstIdleTask = NewCancelableRunnableFunction(FirstIdle);
     gFirstIdleTask = firstIdleTask;
-    NS_IdleDispatchToCurrentThread(firstIdleTask.forget());
+    if (NS_FAILED(NS_IdleDispatchToCurrentThread(firstIdleTask.forget()))) {
+      gFirstIdleTask = nullptr;
+      hasRunOnce = false;
+    }
   }
 
   return nsIContentChild::RecvPBrowserConstructor(aActor,
@@ -2397,6 +2400,7 @@ ContentChild::ActorDestroy(ActorDestroyReason why)
 #else
   if (gFirstIdleTask) {
     gFirstIdleTask->Cancel();
+    gFirstIdleTask = nullptr;
   }
 
   nsHostObjectProtocolHandler::RemoveDataEntries();
